### PR TITLE
return binop cast behavior on +-/*^log

### PR DESCRIPTION
**Pull request by jmcardon**
This PR removes our decision to remove the casting behavior on binary operators for decimals

* [x] Test coverage for the proposed change (See: ops.repl)

### CODE REVIEW

**Review by emilypi — COMMENTED** [review]
It's kind of wonky that we upcast `*` to be decimal-typed in every case except integer args. Do we want to emit a deprecation warning for the March service date deprecating this behavior?

**Review by emilypi — APPROVED** [review]

**Comment by jmcardon** [comment]
> It's kind of wonky that we upcast * to be decimal-typed in every case except integer args. Do we want to emit a deprecation warning for the March service date deprecating this behavior?

Indeed. I will do so in a followup.

File: pact-core/Pact/Core/IR/Eval/CoreBuiltin.hs [hunk 1]
```diff
@@ -148,20 +153,49 @@ rawAdd info b cont handler _env = \case
...
+  [VLiteral (LDecimal i), VLiteral (LDecimal i')] ->
+    decimalSub i i'
+  [VLiteral (LInteger i), VLiteral (LDecimal i')] ->
+    decimalSub (Decimal 0 i) i'
+  [VLiteral (LDecimal i), VLiteral (LInteger i')] ->
+    decimalSub i (Decimal 0 i')
   args -> argsError info b args
+  where
+  decimalSub i i' = do
+    chargeGasArgs info (GIntegerOpCost PrimOpAdd (decimalMantissa i) (decimalMantissa i'))
```
Comment (0xd34df00d) [inline pact-core/Pact/Core/IR/Eval/CoreBuiltin.hs:176]: Should this be `PrimOpAdd`? Addition and subtraction are indeed very similar, so if that's intended, I'd probably rename `PrimOpAdd` → `PrimOpAddSub`.

**Review by 0xd34df00d — APPROVED** [review]

File: pact-core/Pact/Core/IR/Eval/CoreBuiltin.hs (hunk 1 above)
Comment (jmcardon) [reply pact-core/Pact/Core/IR/Eval/CoreBuiltin.hs:176]: Great suggestion!! Will do

**Pull request merged**